cephfs/cephfs_test.go

```diff
@@ -200,6 +221,225 @@ func TestChown(t *testing.T) {
+	assert.NotNil(t, mount)
+
+	err = mount.ReadDefaultConfigFile()
+	assert.NoError(t, err)
+
+	err = mount.SetConf(option, value)
+	assert.NoError(t, err)
+
+	newValue, err := mount.GetConf(option)
+	assert.NoError(t, err)
+
+	assert.Equal(t, value, newValue)
+}
+
+func TestOpenClose(t *testing.T) {
+	mount, err := cephfs.CreateMount("")
+	assert.NoError(t, err)
+	assert.NotNil(t, mount)
+
+	err = mount.ReadDefaultConfigFile()
+	assert.NoError(t, err)
+
+	err = mount.Mount("/")
+	assert.NoError(t, err)
+
+	fd, err := mount.Open("/text.txt", os.O_CREATE, 0755)
+	assert.NoError(t, err)
+
+	err = mount.Close(fd)
+	assert.NoError(t, err)
+
+	err = mount.Unlink("/text.txt")
+	assert.NoError(t, err)
+}
+
+func TestWriteRead(t *testing.T) {
+	mount, err := cephfs.CreateMount("")
+	assert.NoError(t, err)
+	assert.NotNil(t, mount)
+
+	err = mount.ReadDefaultConfigFile()
+	assert.NoError(t, err)
+
+	err = mount.Mount("/")
+	assert.NoError(t, err)
+
+	fd, err := mount.Open("/text.txt", os.O_CREATE|os.O_RDWR, 0755)
+	assert.NoError(t, err)
+
+	data := []byte("Ceph uniquely delivers object, block, and file storage in one unified system.")
+
+	size, err := mount.Write(fd, data, uint64(len(data)), 0)
+	assert.NoError(t, err)
+	assert.Equal(t, size, len(data))
+
+	readData, err := mount.Read(fd, uint64(len(data)), 0)
+	assert.NoError(t, err)
+	assert.Equal(t, data, readData)
+
+	err = mount.Close(fd)
+	assert.NoError(t, err)
+
+	err = mount.Unlink("/text.txt")
+	assert.NoError(t, err)
+}
+
+func TestListDir(t *testing.T) {
+	mount, err := cephfs.CreateMount("")
+	assert.NoError(t, err)
+	assert.NotNil(t, mount)
+
+	err = mount.ReadDefaultConfigFile()
+	assert.NoError(t, err)
+
+	err = mount.Mount("/")
+	assert.NoError(t, err)
+
+	err = mount.MakeDir("/testdir", 0755)
+	assert.NoError(t, err)
+
+	assert.NoError(t, err)
+	l, err := mount.ListDir("/")
+	assert.NoError(t, err)
+	assert.NotNil(t, l)
+
+	err = mount.RemoveDir("/testdir")
+	assert.NoError(t, err)
+}
+
+func TestStatLink(t *testing.T) {
+	mount, err := cephfs.CreateMount("")
+	assert.NoError(t, err)
+	assert.NotNil(t, mount)
+
+	err = mount.ReadDefaultConfigFile()
+	assert.NoError(t, err)
+
+	err = mount.Mount("/")
+	assert.NoError(t, err)
+
+	fd, err := mount.Open("/text.txt", os.O_CREATE, 0755)
+	assert.NoError(t, err)
+	err = mount.Close(fd)
+	assert.NoError(t, err)
+
+	stat, err := mount.LStat("/text.txt")
+	assert.Equal(t, stat.IsFile, true)
+
+	err = mount.MakeDir("/testdir", 0755)
+	assert.NoError(t, err)
+	stat, err = mount.Stat("/testdir")
+	assert.Equal(t, stat.IsDir, true)
+
+	err = mount.Link("/text.txt", "/hardLink")
+	assert.NoError(t, err)
+	stat, err = mount.LStat("/hardLink")
+	assert.Equal(t, stat.IsSymlink, false)
+
+	err = mount.Symlink("/text.txt", "/link")
+	assert.NoError(t, err)
+	stat, err = mount.Stat("/link")
+	assert.Equal(t, stat.IsSymlink, false)
+	stat, err = mount.LStat("/link")
+	assert.Equal(t, stat.IsSymlink, true)
+
+	/*
+	   name, err := mount.ReadLink("/link")
```

Comment:
ReadLink needs a test case too. Did you mean to uncomment this?

Reply:
Hi, @noahdesu , thanks for you review. Yes, I create the ReadLink method and try to test it. However, when I call `ceph_readlink` from C, it returns ** EBADF **. I am trying to figure out why, but I have no idea yet. Do you have any idea?